### ISSUE TITLE
Issue links in changelog

### ISSUE THREAD

**Issue by samestep** [cloverage/cloverage#179]
The guidelines in [CHANGELOG.md] and [CONTRIBUTING.md] say

> Include the issue number (#xxx) which will link back to the originating issue in github.

but it looks like the issue numbers in the changelog do not link to their corresponding issues. Should this sentence be removed from those documents? And should explicit issue links be included in the changelog?

[changelog.md]: https://github.com/cloverage/cloverage/blob/41d16b0f8419c0c6dda05b85e6198e77a8f813c0/CHANGELOG.md#cloverage-changelog
[contributing.md]: https://github.com/cloverage/cloverage/blob/eb44df257eddedc30306e09bfc63794f38ff2705/CONTRIBUTING.md#change-log

**Comment by rm-hull** [comment]
You're right - it doesn't provide a hyperlink (which I think is/was the original intention), but by at least putting the issue number it provides a manual reference back to the issue. Maybe the wording should be changed to:

> Include the issue number (#xxx) which will _reference_ back to the originating issue in github.

What do you think?

**Comment by samestep** [comment]
It [looks like][so] that's just a feature of GitHub Flavored Markdown, which is not used in Markdown files in the repository itself.

You could do that, or you could change it to say to add a Markdown link after running this:

```clojure
(require '[clojure.edn :as edn]
         '[clojure.string :as str])

(def regex #"#(\d+)")

(def url-base "https://github.com/cloverage/cloverage/issues/")

(defn linkify [changelog]
  (->> (re-seq regex changelog)
       (map (comp edn/read-string second))
       (into (sorted-set))
       (map #(str "[i" % "]: " url-base %))
       (#(interleave % (repeat "\n")))
       (apply str)
       (str (str/replace changelog regex "[$0][i$1]") "\n")))

(def filename "CHANGELOG.md")

(->> filename slurp linkify (spit filename))
```

[so]: https://stackoverflow.com/q/16539687/5044950
